src/postamble.js

```diff
@@ -167,7 +167,16 @@ function callMain(args) {
     // that if we get here main returned zero.
     var ret = 0;
 #else
+#if ASYNCIFY == 2
+    // Keep the runtime alive until the entry function promise has resolved.
+    // In ASYNCIFY == 1 mode, this is not needed because it calls an
+    // extra runtimeKeepalivePush when suspending to prevent the maybeExit call
+    // from exiting after resuming.
+    {{{ runtimeKeepalivePush(); }}}
+#endif // ASYNCIFY == 2
```

Comment:
Maybe put this up on line 163 outside he if STANDALONE_WASM?  (unless there is some reason it only have it in this else block?